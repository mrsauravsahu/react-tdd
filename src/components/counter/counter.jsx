import React, { useEffect, useState } from 'react';
import { getNumber } from '../../utils/randomize';
import styles from './counter.module.css';

export const Counter = () => {
  const [count, setCount] = useState(0);

  useEffect(async () => {
    const randomNumber = await getNumber();
    setCount(randomNumber);
  }, []);

  return (
    <div className={styles.container}>
      <div className={styles.message}>{`Current count is ${count}`}</div>
      <button
        className={styles.incrementButton}
        type="button"
        onClick={() => setCount(count + 1)}
      >
        Increment Counter
      </button>
    </div>
  );
};
