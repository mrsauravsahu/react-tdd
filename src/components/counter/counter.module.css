.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: rgb(163, 214, 197);
  padding: 2rem;
  border-radius: 4rem;
}

.message {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 2rem;
  font-weight: 800;
}

.incrementButton {
  margin: 4rem 0 0 0;
  padding: 1rem;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  border: 1px solid green;
  border-radius: 2rem;
  color: white;
  font-size: 1rem;
  font-weight: 800;
  background-color: rgb(0, 103, 0);
  cursor: pointer;
}
