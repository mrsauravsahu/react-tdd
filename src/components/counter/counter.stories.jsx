import React from 'react';
import { Counter } from './counter';

export default {
  component: Counter,
  title: 'components/Counter',
};

export const Default = () => (
  <Counter />
);
