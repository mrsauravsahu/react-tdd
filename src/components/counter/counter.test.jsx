import {
  fireEvent, render, screen,
} from '@testing-library/react';
import React from 'react';
import { Counter } from './counter';

describe(Counter.name, () => {
  test('should display the current count', () => {
    render(<Counter />);

    screen.getByText('Current count is 0');
  });

  test('should display an increment button', () => {
    render(<Counter />);

    const incrementElement = screen.getByText('Increment Counter');
    expect(incrementElement.tagName).toBe('BUTTON');
  });

  test('should update the message when increment button is clicked', () => {
    render(<Counter />);

    const incrementElement = screen.getByText('Increment Counter');
    fireEvent.click(incrementElement);

    screen.getByText('Current count is 1');
  });
});
