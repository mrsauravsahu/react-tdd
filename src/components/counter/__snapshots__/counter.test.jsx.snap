// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="message"
    >
      Current count is 0
    </div>
    <button
      class="incrementButton"
      type="button"
    >
      Increment Counter
    </button>
  </div>
</div>
`;
